File: _203_RemoveLinkedListElements.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeElements(ListNode head, int val) {
        ListNode prev;
        ListNode curr;
        ListNode newHead = head;

        while (newHead != null) {
            if (newHead.val != val) {
                break;
            }
            newHead = newHead.next;
        }
        if (newHead == null) {
            return newHead;
        }
        prev = curr = newHead;

        while (curr != null) {
            if (curr.val != val) {
                prev = curr;
                curr = curr.next;
            } else {
                prev.next = curr.next;
                curr = curr.next;
            }
        }
        return newHead;
    }
}

File: _203_RemoveLinkedListElements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        dummyHead = ListNode()
        dummyHead.next = head
        node = dummyHead
        while head:
            if head.val == val:
                node.next = head.next
            else:
                node.next = head
                node = node.next
            head = head.next
        return dummyHead.next